iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/udf/utils/UDFDataTypeTransformer.java

```diff
@@ -122,19 +126,54 @@ public static org.apache.tsfile.read.common.type.Type transformUDFDataTypeToRead
       case BOOLEAN:
         return BooleanType.BOOLEAN;
       case INT32:
-      case DATE:
         return IntType.INT32;
+      case DATE:
+        return DateType.DATE;
       case INT64:
-      case TIMESTAMP:
         return LongType.INT64;
+      case TIMESTAMP:
+        return TimestampType.TIMESTAMP;
       case FLOAT:
         return FloatType.FLOAT;
       case DOUBLE:
         return DoubleType.DOUBLE;
       case TEXT:
+        return BinaryType.TEXT;
       case BLOB:
+        return BlobType.BLOB;
       case STRING:
+        return StringType.STRING;
+      default:
+        throw new IllegalArgumentException("Invalid input: " + type);
+    }
+  }
+
+  public static org.apache.tsfile.read.common.type.Type transformTSDataTypeToReadType(
```

Comment:
delete this method, you can use TypeFactory.getType instead

Reply:
also delete transformReadTypeToTSDataType in this class, it's not used, and we also have a replacement method called InternalTypeManager.getTSDataType

Reply:
done